plugin/EstateList.php

```diff
@@ -656,6 +661,86 @@ public function getEstateIds(): array
 		return array_column($this->_records, 'id');
 	}
 
+	/**
+	 * @param array $parkingArray
+	 * @param string $language
+	 * @param string $locate
+	 * @return array
+	 */
+	public function formatParkingLot(array $parkingArray, string $language, string $locate = 'de_DE'): array
+	{
+		$messages = [];
+		foreach ($parkingArray as $key => $parking) {
+			if (!$parking['Count']) {
+				continue;
+			}
+			switch ($language) {
+				case 'ENG':
+					$element = "{$parking['Count']} {$this->getParkingName($key, $parking['Count'])} at {$this->formatPrice($parking['Price'], $language, $locate)}";
+					break;
+				default:
+					$element = "{$parking['Count']} {$this->getParkingName($key, $parking['Count'])} à {$this->formatPrice($parking['Price'], $language, $locate)}";
+					break;
+			}
+			if (!empty($parking['MarketingType'])) {
+				$element .= " ({$parking['MarketingType']})";
+			}
+			array_push($messages, $element);
+
+		}
+		return $messages;
+	}
+
+	/**
+	 * @param string $str
+	 * @param string $language
+	 * @param string $locate
+	 * @return string
+	 */
+	public function formatPrice(string $str, string $language, string $locate): string
+	{
+		$digit = intval(substr(strrchr($str, "."), 1));
+		try {
+			$format = new NumberFormatter($locate, NumberFormatter::CURRENCY);
+			if ($digit) {
+				$format->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, 2);
+			} else {
+				$format->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, 0);
+			}
+			return str_replace("\xc2\xa0", " ", $format->formatCurrency($str, "EUR"));;
```

Comment:
is the `str_replace` needed here?

Reply:
Because `NumberFormatter` `formatCurrency` will return the character [NBSP], it will not pass when running the unit test. I need replace [NBSP] with space